src/legacy/ui/public/visualize/loader/pipeline_helpers/build_pipeline.ts

```diff
@@ -104,43 +111,42 @@ export const getSchemas = (vis: Vis, timeRange?: any): Schemas => {
   };
 
   const createSchemaConfig = (accessor: number, agg: AggConfig): SchemaConfig => {
-    const schema = {
-      accessor,
-      format: {},
-      params: {},
-      aggType: agg.type.name,
-    };
-
     if (agg.type.name === 'date_histogram') {
       agg.params.timeRange = timeRange;
       setBounds(agg, true);
     }
-    if (agg.type.name === 'geohash_grid') {
-      schema.params = {
-        precision: agg.params.precision,
-        useGeocentroid: agg.params.useGeocentroid,
-      };
-    }
 
-    if (
-      [
-        'derivative',
-        'moving_avg',
-        'serial_diff',
-        'cumulative_sum',
-        'sum_bucket',
-        'avg_bucket',
-        'min_bucket',
-        'max_bucket',
-      ].includes(agg.type.name)
-    ) {
-      const subAgg = agg.params.customMetric || agg.aggConfigs.byId[agg.params.metricAgg];
-      schema.format = createFormat(subAgg);
-    } else {
-      schema.format = createFormat(agg);
+    const hasSubAgg = [
+      'derivative',
+      'moving_avg',
+      'serial_diff',
+      'cumulative_sum',
+      'sum_bucket',
+      'avg_bucket',
+      'min_bucket',
+      'max_bucket',
+    ].includes(agg.type.name);
+
+    const format = createFormat(
+      hasSubAgg ? agg.params.customMetric || agg.aggConfigs.byId[agg.params.metricAgg] : agg
+    );
+
+    const params: SchemaConfigParams = {
+      isDate: format.id === 'date',
+      isNumeric: format.id === 'number',
+    };
```

Comment:
Lines 134-137 are the ones that actually "fix" the bug -- everything else I changed was simply refactoring for clarity.